xarray/conventions.py

```diff
@@ -602,6 +629,17 @@ def maybe_encode_dtype(var, name=None):
     return var
 
 
+def maybe_encode_bools(var, needs_copy=True):
+    if ((var.dtype == np.bool) and
+            ('dtype' not in var.encoding) and ('dtype' not in var.attrs)):
+        dims, data, attrs, encoding = _var_as_tuple(var)
+        attrs['dtype'] = 'bool'
+        data = data.astype(dtype='i1', copy=needs_copy)
```

Comment:
It looks like casting to `i1` always makes a copy, even though I don't think it's strictly necessary. So let's just drop the needs_copy business here and always copy:
```
In [54]: x
Out[54]: array([ True, False,  True], dtype=bool)

In [55]: y
Out[55]: array([1, 0, 1], dtype=int8)

In [56]: x[0] = False

In [57]: y
Out[57]: array([1, 0, 1], dtype=int8)
```